merge-sort/merge.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import unicode_literals
+from timeit import timeit
+
+
+def merge_sort(lst):
+    if len(lst) <= 1:
+        return lst
+    left = []
+    right = []
+    middle = len(lst) // 2
+    for x in lst[middle:]:
+        left.append(x)
+    for x in lst[:middle]:
+        right.append(x)
+
+    left = merge_sort(left)
+    right = merge_sort(right)
+
+    return merge(left, right)
+
+
+def merge(left, right):
+    result = []
+    while len(left) > 0 and len(right) > 0:
+        if left[0] <= right[0]:
+            result.append(left[0])
+            left = left[1:]
```

Comment:
This is an interesting approach to avoiding the `list.pop(0)` call.  I'm not sure that making a slice of the entire list starting from 1 is much cheaper than pop(0), though.  Might need to do some research on that.